File: src/main/java/org/micoli/php/service/TaskScheduler.java
package org.micoli.php.service;

import com.intellij.util.concurrency.AppExecutorUtil;
import java.util.concurrent.TimeUnit;
import javax.swing.*;

public class TaskScheduler {
    public static void scheduleLater(Runnable runnable, int delayMillis) {
        AppExecutorUtil.getAppScheduledExecutorService()
                .schedule(() -> SwingUtilities.invokeLater(runnable), delayMillis, TimeUnit.MILLISECONDS);
    }
}

File: src/main/kotlin/org/micoli/php/service/TaskScheduler.kt
package org.micoli.php.service

import com.intellij.util.concurrency.AppExecutorUtil
import java.util.concurrent.TimeUnit
import javax.swing.SwingUtilities

object TaskScheduler {
    fun scheduleLater(runnable: Runnable?, delayMillis: Int) {
        AppExecutorUtil.getAppScheduledExecutorService()
            .schedule(
                { SwingUtilities.invokeLater(runnable) },
                delayMillis.toLong(),
                TimeUnit.MILLISECONDS)
    }
}
